control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1584,6 +1601,27 @@ func (r *reconciler) ensureIngressControllersRemoved(ctx context.Context) (bool,
 		return false, fmt.Errorf("failed to force delete pods under openshift-ingress namespace: %w", errors.NewAggregate(errs))
 	}
 
+	// Remove ingress service and route that were created by HCCO
+	if !r.externalInfra {
+		// External infra is not being used, the ingress route and service will be deleted as a result
+		// of removal of the hosted control plane namespace
+		return false, nil
+	}
```

Comment:
there will still be a branching logic in this case, because the client and the namespace are different in case of local infra and external infra.
the reasoning here was to avoid executing code that won't change the outcome.

Reply:
done.